integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -16,28 +16,66 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 import { expect } from "chai";
-import Realm from "realm";
+import Realm, { Mixed, ObjectSchema } from "realm";
 
 import { importAppBefore, authenticateUserBefore, openRealmBefore } from "../../hooks";
 
 import { itUploadsDeletesAndDownloads } from "./upload-delete-download";
 import { buildAppConfig } from "../../utils/build-app-config";
 
-type MixedClass = {
-  _id: Realm.BSON.ObjectId;
-  value: Realm.Mixed;
-  list: Realm.List<Realm.Mixed>;
-};
 type Value = Realm.Mixed | ((realm: Realm) => Realm.Mixed);
-type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void | boolean;
+type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void;
+
+class MixedClass extends Realm.Object<MixedClass> {
+  _id!: Realm.BSON.ObjectId;
+  value: Realm.Mixed;
+  list!: Realm.List<Realm.Mixed>;
+  dict!: Realm.Dictionary<Realm.Mixed>;
+
+  static schema: ObjectSchema = {
+    name: "MixedClass",
+    properties: {
+      _id: "objectId",
+      value: "mixed",
+      list: "mixed[]",
+      dict: "mixed{}",
+    },
+    primaryKey: "_id",
+  };
+}
 
 /**
  * The default tester of values.
  * @param actual The value downloaded from the server.
  * @param inserted The value inserted locally before upload.
  */
 function defaultTester(actual: Realm.Mixed, inserted: Realm.Mixed) {
-  expect(actual).equals(inserted);
+  if (actual instanceof Realm.List) {
```

Comment:
If `actual` is the wrong type (the database is broken), then this might be asserting the wrong thing and we cannot rely on the type of `inserted` since that's has the unmanaged value 🤔 I believe this was why the test functions were separate. Also it allows for more detailed testing and multiple tests per type.

Reply:
Makes sense. I'll reinstantiate the local test functions that were used in the tests and keep this for the mixed testing

Reply:
Did you address this? The current use of this still feel wrong to be, as the assertion performed depends on the type of the `actual`.

Reply:
I think I've fixed this. I've made the testing a little bit more complicated, but now the default tester actually uses the `inserted` value as the truth value